### ISSUE TITLE
CMake Error - Mac OsX High Sierra

### ISSUE THREAD

**Issue by neboduus** [rgburke/grv#34]
Hi all,

I was trying to install grv but i couldn't figure it out.
I followed the instructions from README.md that are:
```
brew install readline pkg-config cmake
go get -d github.com/rgburke/grv/cmd/grv

# I then moved the files from go directory (didn't seem a good directory to me to do <make install>)
mv $GOPATH/src/github.com/rgburke/grv /usr/local/ 
cd /usr/local

# I have errors at this step
make install
```

And the error are:

```
git submodule update --init --recursive
if patch --dry-run -N -d ./cmd/grv/vendor/gopkg.in/libgit2/git2go.v25 -p1 < git2go.v25.patch >/dev/null; then \
		patch -d ./cmd/grv/vendor/gopkg.in/libgit2/git2go.v25 -p1 < git2go.v25.patch; \
	fi
cd ./cmd/grv/vendor/gopkg.in/libgit2/git2go.v25 && git submodule update --init;
make -C ./cmd/grv/vendor/gopkg.in/libgit2/git2go.v25 install-static
./script/build-libgit2-static.sh
+ VENDORED_PATH=vendor/libgit2
+ cd vendor/libgit2
+ mkdir -p install/lib
+ mkdir -p build
+ cd build
+ cmake -DTHREADSAFE=ON -DBUILD_CLAR=OFF -DBUILD_SHARED_LIBS=OFF -DCMAKE_C_FLAGS=-fPIC -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_INSTALL_PREFIX=../install -DUSE_EXT_HTTP_PARSER=OFF -DUSE_HTTPS=OFF -DUSE_NSEC=OFF -DUSE_SSH=OFF -DCURL=OFF ..
-- The C compiler identification is AppleClang 9.0.0.9000039
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -- broken
CMake Error at /usr/local/Cellar/cmake/3.10.2/share/cmake/Modules/CMakeTestCCompiler.cmake:52 (message):
  The C compiler

    "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: /usr/local/go/github.com/rgburke/grv/cmd/grv/vendor/gopkg.in/libgit2/git2go.v25/vendor/libgit2/build/CMakeFiles/CMakeTmp
    
    Run Build Command:"/usr/bin/make" "cmTC_6c550/fast"
    /Applications/Xcode.app/Contents/Developer/usr/bin/make -f CMakeFiles/cmTC_6c550.dir/build.make CMakeFiles/cmTC_6c550.dir/build
    Building C object CMakeFiles/cmTC_6c550.dir/testCCompiler.c.o
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc   -fPIC  -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk   -o CMakeFiles/cmTC_6c550.dir/testCCompiler.c.o   -c /usr/local/go/github.com/rgburke/grv/cmd/grv/vendor/gopkg.in/libgit2/git2go.v25/vendor/libgit2/build/CMakeFiles/CMakeTmp/testCCompiler.c
    Linking C executable cmTC_6c550
    /usr/local/Cellar/cmake/3.10.2/bin/cmake -E cmake_link_script CMakeFiles/cmTC_6c550.dir/link.txt --verbose=1
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -fPIC  -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk -Wl,-search_paths_first -Wl,-headerpad_max_install_names -X 'main.version=v0.1.1-9-gc5bded9' -X 'main.headOid=c5bded9' -X 'main.buildDateTime=2018-02-06 12:35:17 CET'  CMakeFiles/cmTC_6c550.dir/testCCompiler.c.o  -o cmTC_6c550 
    clang: error: no such file or directory: 'main.version=v0.1.1-9-gc5bded9'
    clang: error: no such file or directory: 'main.headOid=c5bded9'
    clang: error: no such file or directory: 'main.buildDateTime=2018-02-06 12:35:17 CET'
    make[3]: *** [cmTC_6c550] Error 1
    make[2]: *** [cmTC_6c550/fast] Error 2
    

  

  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:14 (PROJECT)


-- Configuring incomplete, errors occurred!
See also "/usr/local/go/github.com/rgburke/grv/cmd/grv/vendor/gopkg.in/libgit2/git2go.v25/vendor/libgit2/build/CMakeFiles/CMakeOutput.log".
See also "/usr/local/go/github.com/rgburke/grv/cmd/grv/vendor/gopkg.in/libgit2/git2go.v25/vendor/libgit2/build/CMakeFiles/CMakeError.log".
make[1]: *** [build-libgit2] Error 1
make: *** [build-libgit2] Error 2
```

Note that I already did 
```
brew install readline pkg-config cmake
```
And everything was OK. Only readline installation gave me this message:
```
==> Pouring readline-7.0.3_1.high_sierra.bottle.tar.gz
==> Caveats
This formula is keg-only, which means it was not symlinked into /usr/local,
because macOS provides the BSD libedit library, which shadows libreadline.
In order to prevent conflicts when programs look for libreadline we are
defaulting this GNU Readline installation to keg-only..

For compilers to find this software you may need to set:
    LDFLAGS:  -L/usr/local/opt/readline/lib
    CPPFLAGS: -I/usr/local/opt/readline/include
```
I have also set that EVN VARS in ~/.bash_profile but the error persists.
I someone could help I would be very grateful. Thankx!

**Comment by neboduus** [comment]
I even tried to `make install` in ../go/src folder but the outcome is the same.

**Comment by Jahhein** [comment]
Can you provide these in a gist perhaps?

```log
See also "/usr/local/go/github.com/rgburke/grv/cmd/grv/vendor/gopkg.in/libgit2/git2go.v25/vendor/libgit2/build/CMakeFiles/CMakeOutput.log".
See also "/usr/local/go/github.com/rgburke/grv/cmd/grv/vendor/gopkg.in/libgit2/git2go.v25/vendor/libgit2/build/CMakeFiles/CMakeError.log".
```

I managed to compile this without go get. I simply cloned the repository in my personal scripts bin, did a ``make install`` and symlinked the binary to into my path.

**Comment by neboduus** [comment]
Here are the gists:
1)CMakeOutput.log
https://gist.github.com/neboduus/37b94fc467c832dff32cf23a76922591
1)CMakeError.log
https://gist.github.com/neboduus/7103954dc5d160843641ead3625fec47

Any way I will try now to do it as you did

**Comment by Jahhein** [comment]
I did it the proper way and it completely compiles for me.

```shell
brew install readline pkg-config cmake
go get -d github.com/rgburke/grv/cmd/grv
cd $GOPATH/src/github.com/rgburke/grv
make install
```

Since you want it in your path of `/usr/local` might I suggest doing the following...
```shell
cd /usr/local/bin && \ln -ivs $GOPATH/src/github.com/rgburke/grv/grv
```

Reasons being:

1. The directory itself being in `/usr/local` doesn't make since since it won't be in your path still and will still need to be symlinked down a directory or a new `$PATH` location will need to be added.
2. If you remove `$GOPATH/src/github.com/rgburke/grv` for any reason, homebrew will prune the broken symlink to `grv` executable, cleaning up for you.

**Comment by neboduus** [comment]
@Jahhein  Thanks a lot for your assistance, I did as you did. I mean I cloned the repo into usr/local/bin but `make install` went in the same errors. 

Now I see that you got the same errors. Lets hope someone will help us!

**Comment by rgburke** [comment]
Thanks for reporting this issue and for attaching the cmake logs, they were very helpful. I think I know what the issue is and I have applied a small fix to master e80cb8c08753e7b0e785cacdda67ea1b7633d145. Could you try building GRV again using the instructions @Jahhein described and confirm if it's now successful?

```
brew install readline pkg-config cmake
go get -d github.com/rgburke/grv/cmd/grv
cd $GOPATH/src/github.com/rgburke/grv
make
cd /usr/local/bin
ln -ivs $GOPATH/src/github.com/rgburke/grv/grv
```

The issue itself:
GRV's Makefile defines various variables which one of which was
```
LDFLAGS=-X 'main.version=$(VERSION)' -X 'main.headOid=$(HEAD_OID)' -X 'main.buildDateTime=$(BUILD_DATETIME)'
```
Here we were using `LDFLAGS` to store some values we wanted to pass to `go build` during compilation. However `LDFLAGS` is an implicit make variable used to pass extra arguments to the linker. Therefore when cmake was attempting to find a C compiler it could use it was passing in these linker arguments which are invalid (they only make sense to `go build`).
On my Mac the compiler used is `/Library/Developer/CommandLineTools/usr/bin/cc` which for some reason doesn't seem to be affected by this. This is why GRV has built successfully on other Macs.

**Comment by neboduus** [comment]
Hi @rgburke! After your fixx everything went well 👍  and I am finally able to use `grv`. 
Congrats for your tempestive intervention and for this awesome tool! I can not wait to use it on a real project!
